sw/host/opentitanlib/src/io/jtag.rs

```diff
@@ -131,31 +131,225 @@ pub enum JtagTap {
     LcTap,
 }
 
-/// List of useful RISC-V general purpose registers
-#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
-pub enum RiscvGpr {
-    GP,
-    SP,
-    RA,
+macro_rules! def_enum_name {
+    ($(#[$attr:meta])* pub enum $name:ident {
+        $($reg:ident,)*
+    }) => {paste::paste!{
+        $(#[$attr])*
+        pub enum $name {
+            $($reg,)*
+        }
+
+        impl $name {
+            pub fn name(self) -> &'static str {
+                match self {
+                    $(Self::$reg => stringify!([<$reg:lower>]),)*
+                }
+            }
+        }
+    }}
```

Comment:
Would you mind documenting this macro?

Reply:
Added an example expansion and also an unit test to test two example outputs of the generated `name` function.